rust/lancedb/src/remote/client.rs

```diff
@@ -14,42 +14,207 @@
 
 use std::{future::Future, time::Duration};
 
+use log::debug;
 use reqwest::{
     header::{HeaderMap, HeaderValue},
-    RequestBuilder, Response,
+    Request, RequestBuilder, Response,
 };
 
 use crate::error::{Error, Result};
 
+const REQUEST_ID_HEADER: &str = "x-request-id";
+
+/// Configuration for the LanceDB Cloud HTTP client.
+#[derive(Debug)]
+pub struct ClientConfig {
+    pub timeout_config: TimeoutConfig,
+    pub retry_config: RetryConfig,
+    /// User agent to use for requests. The default provides the libary
+    /// name and version.
+    pub user_agent: String,
+    // TODO: how to configure request ids?
+}
+
+impl Default for ClientConfig {
+    fn default() -> Self {
+        Self {
+            timeout_config: TimeoutConfig::default(),
+            retry_config: RetryConfig::default(),
+            user_agent: concat!("LanceDB-Rust-Client/{}", env!("CARGO_PKG_VERSION")).into(),
```

Comment:
Should this be `format!` instead of `concat!`, or otherwise we remove the `{}`?
```rs
fn main() {
    let test = concat!("test/{}", "test2");
    println!("{}", test);
}
```
prints: `test/{}test2`